enterprise_subsidy/apps/api/v1/views/subsidy.py

```diff
@@ -74,6 +78,32 @@ class SubsidyViewSet(
     allowed_roles = [ENTERPRISE_SUBSIDY_ADMIN_ROLE, ENTERPRISE_SUBSIDY_OPERATOR_ROLE]
     role_assignment_class = EnterpriseSubsidyRoleAssignment
 
+    def handle_exception(self, exc):
+        response = super().handle_exception(exc)
+
+        if isinstance(exc, ServerError):
+            response.data = exc.get_full_details()
+        elif response is not None and "detail" in response.data:
+            response.data["error_code"] = responses.get(response.status_code, "unknown_error")
+            response.data["developer_message"] = response.data["detail"]
+            response.data["user_message"] = response.data["detail"]
+            del response.data["detail"]
+        return response
+
+    def get_serializer_class(self, *args, **kwargs):
+        """
+        Return the serializer class to use for the current request.
+
+        We override the function here instead of just setting the ``serializer_class`` class attribute because that
+        attribute only supports using a single serializer for the entire viewset.  This override logic allows for the
+        serializer class to be based conditionally on the type of action.
+        """
+        if self.request_action.lower() in ('put', 'patch'):
+            return SubsidyUpdateRequestSerializer
```

Comment:
It's odd that this line of code isn't covered by tests...

Reply:
I would double-check in the debugger that this line of code is actually executed when running tests.

Reply:
You're right, the breakpoint showed that this code wasn't running! The breakpoint worked if I change it to `self.request.method.lower()`.